modules/services/window-managers/i3-sway/sway.nix

```diff
@@ -0,0 +1,311 @@
+    default_floating_border ${if floating.titlebar then "normal" else "pixel"} ${toString floating.border}
+    hide_edge_borders ${window.hideEdgeBorders}
+    focus_wrapping ${if focus.forceWrapping then "yes" else "no"}
+    focus_follows_mouse ${if focus.followMouse then "yes" else "no"}
+    focus_on_window_activation ${focus.newWindow}
+    mouse_warping ${if focus.mouseWarping then "output" else "none"}
+    workspace_layout ${workspaceLayout}
+
+    client.focused ${colorSetStr colors.focused}
+    client.focused_inactive ${colorSetStr colors.focusedInactive}
+    client.unfocused ${colorSetStr colors.unfocused}
+    client.urgent ${colorSetStr colors.urgent}
+    client.placeholder ${colorSetStr colors.placeholder}
+    client.background ${colors.background}
+
+    ${keybindingsStr keybindings}
+    ${keycodebindingsStr keycodebindings}
+    ${concatStringsSep "\n" (mapAttrsToList inputStr input)}
+    ${concatStringsSep "\n" (mapAttrsToList outputStr output)}
+    ${concatStringsSep "\n" (mapAttrsToList modeStr modes)}
+    ${concatStringsSep "\n" (mapAttrsToList assignStr assigns)}
+    ${concatStringsSep "\n" (map barStr bars)}
+    ${optionalString (gaps != null) gapsStr}
+    ${concatStringsSep "\n" (map floatingCriteriaStr floating.criteria)}
+    ${concatStringsSep "\n" (map windowCommandsStr window.commands)}
+    ${concatStringsSep "\n" (map startupEntryStr startup)}
+  '' else "") + "\n" + (if cfg.systemdIntegration then ''
+    exec "systemctl --user import-environment; systemctl --user start sway-session.target"
+  '' else "") + cfg.extraConfig);
+in
+
+{
+  options.services.sway = {
+    enable = mkEnableOption "sway wayland compositor";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sway;
+      defaultText = literalExample "${pkgs.sway}";
+      description = ''
+        Sway package to use.
+      '';
+    };
+
+    systemdIntegration = mkOption {
+      type = types.bool;
+      default = pkgs.stdenv.isLinux;
+      example = false;
+      description = ''
+        Whether to enable sway-session.target on sway startup. This links to graphical-session.target
+      '';
+    };
+
+    xwayland = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable xwayland, which is needed for the default configuration of sway.
+      '';
+    };
+
+    config = mkOption {
+      type = types.nullOr configModule;
+      default = {};
+      description = "Sway configuration options.";
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = "Extra configuration lines to add to ~/.config/sway/config.";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ] ++ lib.optional cfg.xwayland pkgs.xwayland;
+    home.file.".config/sway/config".text = configText;
+    systemd.user.targets.sway-session = mkIf cfg.systemdIntegration {
+      Unit = {
```

Comment:
Shouldn't there be a
```
Requires = [ "graphical-session-pre.target" ];
```
to ensure the pre target kicks off?

Also, do you know why the `Wants` and `After` fields are needed? Should they be added to hm-graphical-session?

Reply:
no idea, I just copied https://github.com/swaywm/sway/wiki/Systemd-integration

Reply:
Would you like me to add that?